backend/api/Controllers/EmergencyActionController.cs

```diff
@@ -0,0 +1,178 @@
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> ClearEmergencyState(
+            [FromRoute] string robotId,
+            [FromRoute] string installationCode,
+            [FromRoute] string areaName)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                _logger.LogWarning("Could not find robot with id {Id}", robotId);
+                return NotFound("Robot not found");
+            }
+
+            var area = await _areaService.ReadByInstallationAndName(installationCode, areaName);
+            if (area == null)
+            {
+                _logger.LogError("Could not find area {AreaName} for installation code {InstallationCode}", areaName, installationCode);
+                return NotFound("Area not found");
+            }
+
+            _emergencyActionService.TriggerEmergencyButtonDepressedForRobot(new EmergencyButtonPressedForRobotEventArgs(robot.Id, area.Id));
+
+            return Ok("Request to clear emergency state for robot was received");
+        }
+
+        /// <summary>
+        ///     This query will clear the emergency state that is introduced by aborting the current mission and returning to a
+        ///     safe zone. Clearing the emergency state means that mission runs that may be in the robots queue will start."
+        /// </summary>
+        [HttpPost]
+        [Route("{robotId}/clear-emergency-state")]
+        [Authorize(Roles = Role.User)]
+        [ProducesResponseType(typeof(MissionRun), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> ClearInstallationEmergencyState(
+            [FromRoute] string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+
+            if (robot == null)
+            {
+                _logger.LogWarning("Could not find robot with id {Id}", robotId);
+                return NotFound("Robot not found");
+            }
+            if (robot.CurrentInstallation == null)
+            {
+                _logger.LogWarning("Could not find installation for robot with id {Id}", robotId);
+                return NotFound("Installation not found");
+            }
+            if (robot.CurrentArea == null)
+            {
+                _logger.LogWarning("Could not find area for robot with id {Id}", robotId);
+                return NotFound("Area not found");
+            }
+
+            _emergencyActionService.TriggerEmergencyButtonDepressedForRobot(new EmergencyButtonPressedForRobotEventArgs(robot.Id, robot.CurrentArea!.Id));
```

Comment:
Is it necessary to have this controller endpoint in addition to the one above? Seems to me that only one should be required.